catlin/pkg/validator/task_validator_test.go

```diff
@@ -0,0 +1,57 @@
+package validator
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/tektoncd/plumbing/catlin/pkg/parser"
+	"gotest.tools/v3/assert"
+)
+
+const taskWithInvalidImageRef = `
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: invalid-image-refs
+  labels:
+    app.kubernetes.io/version: a,b,c
+  annotations:
+    tekton.dev/tags: a,b,c
+    tekton.dev/pipelines.minVersion: "0.12"
+    tekton.dev/displayName: My Example Task
+spec:
+  description: |-
+    A summary of the resource
+  
+    A para about this valid task
+  
+  steps:
+  - name: hello
+    image: ubuntu
+    command: [sleep, infinity]
+  - name: foo
+    image: abc.io/fedora:latest
+  - name: bar
+    image: abc.io/fedora:1.0@sha256:deadb33fdeadb33fdeadb33f
+  - name: valid
+    image: abc.io/ubuntu:1.0
+`
+
+func TestTaskValidator_InvalidImageRef(t *testing.T) {
+
+	r := strings.NewReader(taskWithInvalidImageRef)
+	parser := parser.ForReader(r)
+
+	res, err := parser.Parse()
+	assert.NilError(t, err)
+
+	v := ForKind(res)
+	result := v.Validate()
+
+	assert.Equal(t, 3, result.Errors)
+	assert.Equal(t, 3, len(result.Lints))
+	assert.Equal(t, `Step "hello" has an invalid image: could not parse reference: ubuntu`, result.Lints[0].Message)
```

Comment:
wait, it should be the same error as `abc.io/fedora:latest` here. `ubuntu` is valid (because it's an "official" image), as it would become `docker.io/library/ubuntu` :upside_down_face:

Reply:
`docker.io/library/ubuntu` would be valid but `ubuntu` invalid as per current implementation. Is that okay?

Reply:
@vdemeester `ubunbu` without a tag or a sha is valid?

Reply:
go-containerreg fails to parse reference to `ubuntu` - right? isn't that why we are getting this error?

Reply:
`go-containerregistry/pkg/name.ParseReference` fails to validate the image ref as it does not have registry appended `docker.io`. Is there a way to which know is an official image?

Reply:
IIUC, https://github.com/google/go-containerregistry/blob/093adf2740a9617fbb6e6fe81ec37b00d3eb1b43/pkg/name/digest_test.go would fail validation of `ubuntu`

Reply:
oh :joy_cat: so yeah, `ubuntu` should be valid. It is *always* better to use the full ref (`docker.io/library/ubuntu`) but `ubuntu` should be valid. An *official* image is like `ubuntu`, no username/group or host on top of it. `traefik`, `debian`, `fedora`, `mysql`, … all those are "official" images

Reply:
We could *warn* in case of those, recommending using the full ref..

Reply:
I am amazed `go-containerregistry` fails on those though…

Reply:
Oh, I guess it depends on the `opt.strict` option thouhg…

Reply:
Yeah. If we use `WeakValidation` then it doesn't fail for ubuntu but also for any image name for eg. `api` or `hub`
How do we differentiate between official and non-official images or should we warn for both?

Reply:
So, I am fine using weak validation and have a `hub` image being seen as valid, even though it doesn't exists — as `foo.com/bar/baz:mytag` is gonna be seen as valid even though it doesn't exists.

And, yeah, I would warn for any image that is just *one part*, aka `ubuntu`, `api`, … being "official" or not, with a message "prefer using a fully qualified reference, like `library/ubuntu` or `docker.io/library/ubuntu` instead".

Reply:
okay then Is `tag` required for all images?

Reply:
a tag is required for all images and should be an error. images like `hub` or `ubuntu` should have a warn telling to use a fully qualified reference, like `library/hub`, `docker.io/library/ubuntu`.